services/121-service/src/payments/fsp-integration/intersolve-visa/entities/intersolve-visa-parent-wallet.entity.ts

```diff
@@ -0,0 +1,49 @@
+import { Base121Entity } from '@121-service/src/base.entity';
+import { IntersolveVisaChildWalletEntity } from '@121-service/src/payments/fsp-integration/intersolve-visa/entities/intersolve-visa-child-wallet.entity';
+import { IntersolveVisaCustomerEntity } from '@121-service/src/payments/fsp-integration/intersolve-visa/entities/intersolve-visa-customer.entity';
+import {
+  Column,
+  Entity,
+  Index,
+  JoinColumn,
+  OneToMany,
+  OneToOne,
+  Relation,
+} from 'typeorm';
+
+// TODO: Should this Entity extend from CascadeDeleteEntity instead? When should an Entity extend from it?
+@Entity('intersolve_visa_parent_wallet')
+export class IntersolveVisaParentWalletEntity extends Base121Entity {
+  @OneToOne(() => IntersolveVisaCustomerEntity)
+  @JoinColumn({ name: 'intersolveVisaCustomerId' })
+  public intersolveVisaCustomer: Relation<IntersolveVisaCustomerEntity>;
+  @Column({ type: 'int' })
+  public intersolveVisaCustomerId: number;
+
+  @OneToMany(
+    (_type) => IntersolveVisaChildWalletEntity,
+    (intersolveVisaChildWallets) =>
+      intersolveVisaChildWallets.intersolveVisaParentWallet,
+  )
+  public intersolveVisaChildWallets: Relation<
+    IntersolveVisaChildWalletEntity[]
+  >;
+
+  @Index()
+  @Column({ unique: true })
+  public tokenCode: string;
+
+  @Column({ default: false })
+  public isLinkedToVisaCustomer: boolean;
+
+  @Column({ default: 0 })
+  public balance: number;
+
+  // Last time we got an update from Intersolve about the wallet status or balance or when it was last used
+  @Column()
+  public lastExternalUpdate: Date;
```

Comment:
This used to be nullable. Is it a conscious decision to make it not nullable? E.g because you have to take into account to also 'do' this update during a payment before you save this this entity. Also if that update than fails you are not able to save this wallet and lose the data. @diderikvw  @PeterSmallenbroek

Reply:
We discussed this and it should be nullable